lxd/db/query/dump.go

```diff
@@ -11,89 +11,103 @@ import (
 )
 
 // Dump returns a SQL text dump of all rows across all tables, similar to
-// sqlite3's dump feature
-func Dump(ctx context.Context, tx *sql.Tx, schema string, schemaOnly bool) (string, error) {
-	schemas := dumpParseSchema(schema)
-
-	// Begin
-	dump := `PRAGMA foreign_keys=OFF;
-BEGIN TRANSACTION;
-`
-	// Schema table
-	tableDump, err := dumpTable(ctx, tx, "schema", dumpSchemaTable)
+// sqlite3's dump feature.
+func Dump(ctx context.Context, tx *sql.Tx, schemaOnly bool) (string, error) {
+	tablesSchemas, err := getTablesSchemas(ctx, tx)
 	if err != nil {
-		return "", fmt.Errorf("Failed to dump table schema: %w", err)
+		return "", err
 	}
-	dump += tableDump
 
-	// All other tables
-	tables := make([]string, 0)
-	for table := range schemas {
-		tables = append(tables, table)
+	tableNames := make([]string, len(tablesSchemas))
+	i := 0
+	for tableName := range tablesSchemas {
+		tableNames[i] = tableName
+		i++
 	}
-	sort.Strings(tables)
-	for _, table := range tables {
-		if schemaOnly {
-			// Dump only the schema.
-			dump += schemas[table] + "\n"
-			continue
-		}
 
-		tableDump, err := dumpTable(ctx, tx, table, schemas[table])
-		if err != nil {
-			return "", fmt.Errorf("Failed to dump table %s: %w", table, err)
-		}
+	// Begin dump string.
+	var builder strings.Builder
+	builder.WriteString("PRAGMA foreign_keys=OFF;\n")
+	builder.WriteString("BEGIN TRANSACTION;\n")
+
+	// Sort table names for repeatability.
+	sort.Strings(tableNames)
```

Comment:
Nice :)

Reply:
They were sorted beforehand too, the only difference now is that the `schema` table wont appear at the top of the dump. :+1: